From: SeongJae Park <sj@kernel.org>
Subject: Docs/mm/damon/design: document DAMOS regions walking
Date: Fri, 3 Jan 2025 09:43:58 -0800

DAMOS' regions walking is a feature for efficiently retrieving monitoring
results or DAMOS-internal behavior.  It can be useful for multiple
purposes including investigations and tuning.  Add a section for it on the
design document.

Link: https://lkml.kernel.org/r/20250103174400.54890-9-sj@kernel.org
Signed-off-by: SeongJae Park <sj@kernel.org>
Cc: Jonathan Corbet <corbet@lwn.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 Documentation/mm/damon/design.rst |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/Documentation/mm/damon/design.rst~docs-mm-damon-design-document-damos-regions-walking
+++ a/Documentation/mm/damon/design.rst
@@ -540,6 +540,17 @@ To know how user-space can set the water
 documentation.
 
 
+Regions Walking
+~~~~~~~~~~~~~~~
+
+DAMOS feature allowing users access each region that a DAMOS action has just
+applied.  Using this feature, DAMON :ref:`API <damon_design_api>` allows users
+access full properties of the regions including the access monitoring results.
+:ref:`DAMON sysfs interface <sysfs_interface>` also allows users read the data
+via special :ref:`files <sysfs_schemes_tried_regions>`.
+
+.. _damon_design_api:
+
 Application Programming Interface
 ---------------------------------
 
_
